ols.py

```diff
@@ -1,68 +1,78 @@
 # base python things
+import uuid
 from typing import Union
-from fastapi import FastAPI, HTTPException
-import gradio as gr
-from dotenv import load_dotenv
 from pydantic import BaseModel
-import os
-import uuid
+from fastapi import FastAPI, HTTPException, Request
+
+# auxiliary functionalities
+from src.config import Config
 
 # internal modules
-from modules.task_breakdown import TaskBreakdown
-from modules.task_processor import TaskProcessor
 from modules.question_validator import QuestionValidator
 from modules.yaml_generator import YamlGenerator
 from modules.happy_response_generator import HappyResponseGenerator
 from modules.docs_summarizer import DocsSummarizer
 from modules.model_context import get_watsonx_predictor
 from modules.conversation_cache import LRUCache
-from modules.gradio_ui import ui
-
-# internal tools
-from tools.ols_logger import OLSLogger
 
-load_dotenv()
+# gradio ui
+from modules.gradio_ui import gradioUI
 
-base_completion_model = os.getenv("BASE_COMPLETION_MODEL", "ibm/granite-20b-instruct-v1")
-
-# TODO: env for verbose chains
-
-# TODO: should this class get moved to a separate file?
+# TODO: Move to external file
 class LLMRequest(BaseModel):
     query: str
     conversation_id: Union[str, None] = None
     response: Union[str, None] = None
 
 class FeedbackRequest(BaseModel):
     conversation_id: int # required
-    feedback_object: str # a json blob 
+    feedback_object: str # a json blob
 
-
-conversation_cache=LRUCache(100)
+# common globals
 app = FastAPI()
-gr.mount_gradio_app(app,ui,path="/ui")
 
+config = Config()
+logger = config.logger
+
+# determine if need to enable local UI
+if config.enable_ui:
+    app=gradioUI().mount_ui(app)
+else:
+    logger.info("Local UI is disabled. To enable set OLS_ENABLE_UI=True")
+
+# application global variables
+base_completion_model = config.base_completion_model
+conversation_cache = LRUCache(100)
 
 def get_suid():
     return str(uuid.uuid4().hex)
 
-
 @app.get("/healthz")
+@app.get("/readyz")
 def read_root():
```

Comment:
"read_root"?  can we give this a better name, if it's going to be the healthz/readyz endpoint?

(also those paths/endpoints may not always share the same implementation, but for now i guess it's fine)

Reply:
Yes, I would rather map all these to a status function. For now, moving "as is"